.github/workflows/release.yml

```diff
@@ -0,0 +1,126 @@
+          # We don't want zero-padding. Use space-padding, then delete the spaces.
+          version=$(date -u +'%Y.%_m.%_d.%_H.%_M' | sed 's/ //g')
+          echo "Generated version number: $version"
+          echo "version=$version" >> $GITHUB_OUTPUT
+
+          # Find package root dir
+          if [[ '${{ inputs.package }}' = 'benchadapt' ]]; then
+            pkg_root=./benchadapt/python
+          elif [[ '${{ inputs.package }}' = 'benchclients' ]]; then
+            pkg_root=./benchclients/python
+          elif [[ '${{ inputs.package }}' = 'benchconnect' ]]; then
+            pkg_root=./benchconnect
+          elif [[ '${{ inputs.package }}' = 'benchrun' ]]; then
+            pkg_root=./benchrun/python
+          elif [[ '${{ inputs.package }}' = 'conbench' ]]; then
+            pkg_root=.
+          fi
+          echo "Package root: $pkg_root"
+          echo "pkg_root=$pkg_root" >> $GITHUB_OUTPUT
+
+          # Find version file (standardized for all packages in this repo)
+          version_file="${pkg_root}/${{ inputs.package }}/_version.py"
+          echo "Version file to update: $version_file"
+          echo "version_file=$version_file" >> $GITHUB_OUTPUT
+
+          # Generate a branch name
+          branch_name="benchbot/${{ inputs.package }}-${version}"
+          echo "Branch name: $branch_name"
+          echo "branch_name=$branch_name" >> $GITHUB_OUTPUT
+
+          # Generate a commit message
+          commit_msg="🤖 ci: bump version of ${{ inputs.package }} to $version"
+          echo "Commit message: $commit_msg"
+          echo "commit_msg=$commit_msg" >> $GITHUB_OUTPUT
+      - name: Replace version file
+        run: |
+          echo \
+            '# Do not touch this autogenerated file\n__version__ = "${{ steps.variables.outputs.version }}"' \
+            > ${{ steps.variables.outputs.version_file }}
+      - name: Commit to a branch
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git checkout -B ${{ steps.variables.outputs.branch_name }}
+          git add ${{ steps.variables.outputs.version_file }}
+          git commit -m '${{ steps.variables.outputs.commit_msg }}'
+          git push
+      - name: Open and merge a PR
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const pull = await github.rest.pulls.create({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              title: '${{ steps.variables.outputs.commit_msg }}',
+              head: '${{ steps.variables.outputs.branch_name }}',
+              base: 'main',
+              body: 'Beep, boop. I am a bot, requested to open this PR in preparation to release to PyPI.'
+            })
+
+            await github.rest.pulls.merge({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: pull.number
+            })
```

Comment:
I'm fine to do this in a pull request like this, but we could also just push to main (that's effectively what's going on here since it opens and closes without anything in between), no?

Reply:
Yes we could! I thought the PR might give a little more observability, and create a place if we want discussion. But if it's too much noise we can just push to main.